go-controller/pkg/ovn/policy.go

```diff
@@ -1098,13 +1097,23 @@ func (oc *Controller) createNetworkPolicy(np *networkPolicy, policy *knet.Networ
 		var errs []error
 		selectedPods = objs
 		policyPorts, ingressDenyPorts, egressDenyPorts := oc.processLocalPodSelectorSetPods(policy, np, selectedPods...)
-		pg.Ports = append(pg.Ports, policyPorts...)
-		ops, err = libovsdbops.AddPortsToPortGroupOps(oc.nbClient, ops, portGroupIngressDenyName, ingressDenyPorts...)
+		newOps, err := libovsdbops.AddPortsToPortGroupOps(oc.nbClient, nil, portGroupIngressDenyName, ingressDenyPorts...)
+		if err != nil {
+			oc.processLocalPodSelectorDelPods(np, selectedPods...)
+			errs = append(errs, err)
+		}
+		newOps, err = libovsdbops.AddPortsToPortGroupOps(oc.nbClient, newOps, portGroupEgressDenyName, egressDenyPorts...)
+		if err != nil {
+			oc.processLocalPodSelectorDelPods(np, selectedPods...)
+			errs = append(errs, err)
+		}
+		pg.Ports = policyPorts
+		newOps, err = libovsdbops.CreateOrUpdatePortGroupsOps(oc.nbClient, newOps, pg)
 		if err != nil {
 			oc.processLocalPodSelectorDelPods(np, selectedPods...)
 			errs = append(errs, err)
 		}
-		ops, err = libovsdbops.AddPortsToPortGroupOps(oc.nbClient, ops, portGroupEgressDenyName, egressDenyPorts...)
+		_, err = libovsdbops.TransactAndCheck(oc.nbClient, append(ops, newOps...))
```

Comment:
adding transact call here will require additional db cleanup in the following code in case of error or deleted np, because before transaction was the last operation in this function